sklearn/ensemble/_base.py

```diff
@@ -127,25 +127,25 @@ def __init__(self, base_estimator, *, n_estimators=10, estimator_params=tuple())
         # still change. Eg., when grid-searching with the nested object syntax.
         # self.estimators_ needs to be filled by the derived classes in fit.
 
-    def _validate_estimator(self, default=None):
-        """Check the estimator and the n_estimator attribute.
-
-        Sets the base_estimator_` attributes.
-        """
+    def _validate_n_estimators(self):
+        """Check the n_estimator attribute."""
         if not isinstance(self.n_estimators, numbers.Integral):
             raise ValueError(
-                "n_estimators must be an integer, got {0}.".format(
-                    type(self.n_estimators)
-                )
+                f"n_estimators must be an integer, got {type(self.n_estimators)}."
             )
 
         if self.n_estimators <= 0:
             raise ValueError(
-                "n_estimators must be greater than zero, got {0}.".format(
-                    self.n_estimators
-                )
+                f"n_estimators must be greater than zero, got {self.n_estimators}."
             )
 
+    def _validate_estimator(self, default=None):
+        """Check the estimator and the n_estimator attribute.
+
+        Sets the base_estimator_` attributes.
+        """
+        self._validate_n_estimators()
```

Comment:
I don't think this is worth adding this function because it is called only once.
I would not refactor this part, only introduced the f-string.